src/allmydata/test/test_connections.py

```diff
@@ -71,33 +72,42 @@ def test_launch_executable(self):
             self.assertEqual(f.mock_calls, [exp])
             self.assertIdentical(h, h1)
 
-    def test_socksport(self):
+    def test_socksport_unix_endpoint(self):
+        n = FakeNode(BASECONFIG+"[tor]\nsocks.port = unix:/var/lib/fw-daemon/tor_socks.socket\n")
+        h1 = mock.Mock()
+        with mock.patch("foolscap.connections.tor.socks_endpoint",
+                        return_value=h1) as f:
+            h = n._make_tor_handler()
+            self.assertTrue(IStreamClientEndpoint.providedBy(f.mock_calls[0]))
+            self.assertIdentical(h, h1)
+
+    def test_socks_endpoint(self):
         n = FakeNode(BASECONFIG+"[tor]\nsocks.port = tcp:127.0.0.1:1234\n")
         h1 = mock.Mock()
-        with mock.patch("foolscap.connections.tor.socks_port",
+        with mock.patch("foolscap.connections.tor.socks_endpoint",
                         return_value=h1) as f:
             h = n._make_tor_handler()
-            self.assertEqual(f.mock_calls, [mock.call("127.0.0.1", 1234)])
+            self.assertTrue(IStreamClientEndpoint.providedBy(f.mock_calls[0]))
             self.assertIdentical(h, h1)
 
-    def test_socksport_otherhost(self):
+    def test_socks_endpoint_otherhost(self):
         n = FakeNode(BASECONFIG+"[tor]\nsocks.port = tcp:otherhost:1234\n")
         h1 = mock.Mock()
-        with mock.patch("foolscap.connections.tor.socks_port",
+        with mock.patch("foolscap.connections.tor.socks_endpoint",
                         return_value=h1) as f:
             h = n._make_tor_handler()
-            self.assertEqual(f.mock_calls, [mock.call("otherhost", 1234)])
+            self.assertTrue(IStreamClientEndpoint.providedBy(f.mock_calls[0]))
             self.assertIdentical(h, h1)
 
     def test_socksport_bad_endpoint(self):
```

Comment:
actually, nevermind, these tests should indeed be named "socksport" because that's the name of the tahoe.cfg key. (I was thinking of the name of the foolscap.connections.tor API call, but really that's not as important here)